### PR TITLE
Feature/name email search list users

### PR DESCRIPTION
**Pull request by 2piJareem**
## Checklist
- [x] I'm using the correct PHP Version (8.1).
- [x] I've added comments to any new methods I've created or where else relevant.
- [x] I've replaced magic method usage on DbService classes with the getInstance() static method.
- [x] I've written any documentation for new features or where else relevant in the docs [repo](https://github.com/2pisoftware/cmfive-docs).

## Description
Added Login, Name and Email filter to List Users function within Admin menu.  
Filter fields match any part of the target string.  
"Name" searches first and last names for a match.

## Changelog
- modules/admin/actions/users.php  Added filter functionality
- modules/admin/templates/users.tpl.php  Added filter to template

refs:
issues:

## Other Information

Docs pull request:

### CODE REVIEW

File: system/templates/base/.gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 node_modules/
-dist/
\ No newline at end of file
+dist/
+package-lock.json
```
Comment (adam-buckley) [inline system/templates/base/.gitignore:3]: Remove this and add package-lock.json back in

**Review by adam-buckley — CHANGES_REQUESTED** [review]

File: system/templates/base/.gitignore (hunk 1 above)
Comment (2piJareem) [reply system/templates/base/.gitignore]: Sorry I didn't see this earlier.  when I remove the dist/ folder from .gitignore I get 8 files form the dist folder and below that want to be committed - Do we want to commit these files as well?
Screenshot 2023-07-06 at 3 17 12 pm

**Review by adam-buckley — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 2]
```diff
@@ -9,3 +9,4 @@
 /system/tests/composer.lock
 /system/tests/support/_generated
 system/system
+system/templates/base/package-lock.json
```
[the remark this answers is not in the document]
Comment (2piJareem) [reply .gitignore:12]: Removed

File: system/modules/insights/actions/index.php [hunk 3]
```diff
@@ -48,3 +56,20 @@ function index_ALL(Web $w)
     //send the table to the template using ctx
     $w->ctx('insightTable', Html::table($table, 'insight_table', 'tablesorter', $tableHeaders));
 }
+
+// Function to recursively check if a user is a member of a group (or parent group)
+function checkUserAccess(Web $w, $group, $user_id) : bool {
```
[the remark this answers is not in the document]
Comment (2piJareem) [reply system/modules/insights/actions/index.php:61]: After putting the function into InsightService
eg     public function checkUserAccess(Web $w, $group, $user_id): bool

How does the action function find the function in the service class?  
I get "undefined function" if called as "checkUserAccess(...)"

and "Non static method 'checkUserAccess' should not be called statically."  if I called as "InsightService::checkUserAccess(...)"

[the remark this answers is not in the document]
Comment (2piJareem) [reply system/modules/insights/actions/index.php]: sorted

[the remark this answers is not in the document]
Comment (2piJareem) [reply system/modules/insights/actions/index.php]: btw  should "checkUserAccess(...)" function go in InsightService or AuthService?

**Comment by adam-buckley** [comment]
Moved to draft cause I dont think this PR offers anything anymore. Holding off to make sure soon

**Pull request closed without merging**